### DISCUSSION TITLE
performedOnRaw

### DISCUSSION THREAD

**Discussion by usernotnull** [spatie/laravel-activitylog#1003] [Ideas]
There may be situations where the models are not fetched, but we know in advance their type and id.
If the situation requires many models to be logged, currently we'll have to fetch all the models to be able to call performedOn($model).

How about adding something like 
```
public function performedOnRaw(mixed $primaryKey, string $class): static
{
    ...
}
```

**Comment by Gummibeer** [comment]
In theory that's possible - would need some checks on the `$class` variable (`class_exists`, `is_subclass_of` ...) but feel free to send in a PR.

**Reply by usernotnull** [reply]
Glad to see you're still giving time to OS, Tom (who hasn't read your blog post :) ).
I'll submit a PR this week most probably in that case.

**Reply by Gummibeer** [reply]
Thanks! 🙂
And yes, will check in from time to time. So possibly someone else reviews your PR or it will take time. 🙈
